src/components/link.module.scss

```diff
@@ -0,0 +1,30 @@
+@import '../shared/shared';
+
+.highlight {
+  transition: background-size 0.4s ease;
+  padding-bottom: inherit;
+
+  &:hover {
+    background-size: 100% 100%;
+  }
+}
+
+.pink {
+  @extend .highlight;
```

Comment:
If `.highlight` is never used as an actual CSS class then you can make it a `%` class that's only used by SASS:

```scss
%highlight {
  …
}

.pink {
  @extend %highlight
}
```

https://sass-lang.com/documentation/style-rules/placeholder-selectors